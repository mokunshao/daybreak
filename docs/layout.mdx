---
title: Layout
---

import {useState} from 'react';
import { Playground, Props } from 'docz';
import Layout from '../src/layout/layout.tsx';
import Header from '../src/layout/header.tsx';
import Content from '../src/layout/content.tsx';
import Footer from '../src/layout/footer.tsx';
import Aside from '../src/layout/aside.tsx';

# Layout

```javascript
import { Layout,Header,Content,Footer,Aside } from 'daybreak';
```

## Layout 1

<Playground>
{() => {
  const headerStyle={
    backgroundColor:'#f4f3f3',
    padding:'1em',
  }
  const contentStyle={
    backgroundColor:'#dfdfdf',
    padding:'1em',
  }
  const footerStyle={
    backgroundColor:'#bfd8d5',
    padding:'1em',
  }
  const layoutStyle = {
    height: 300,
    maxWidth: 500,
  }
  return (
    <Layout style={layoutStyle}>
      <Header style={headerStyle}>header</Header>
      <Content style={contentStyle}>content</Content>
      <Footer style={footerStyle}>footer</Footer>
    </Layout>
  )
}}
</Playground>

## Layout 2

<Playground>
{() => {
  const headerStyle={
    backgroundColor:'#f4f3f3',
    padding:'1em',
  }
  const contentStyle={
    backgroundColor:'#dfdfdf',
    padding:'1em',
  }
  const asideStyle={
    backgroundColor:'#bfd8d5',
    padding:'1em',
  }
  const footerStyle = headerStyle;
  const layoutStyle = {
    height: 300,
    maxWidth: 500,
  }
  return (
    <Layout style={layoutStyle}>
      <Header style={headerStyle}>header</Header>
      <Layout>
        <Aside style={asideStyle}>aside</Aside>
        <Content style={contentStyle}>content</Content>
      </Layout>
      <Footer style={footerStyle}>footer</Footer>
    </Layout>
  )
}}
</Playground>

## Layout 3

<Playground>
{() => {
  const headerStyle={
    backgroundColor:'#f4f3f3',
    padding:'1em',
  }
  const contentStyle={
    backgroundColor:'#dfdfdf',
    padding:'1em',
  }
  const asideStyle={
    backgroundColor:'#bfd8d5',
    padding:'1em',
  }
  const footerStyle = headerStyle;
  const layoutStyle = {
    height: 300,
    maxWidth: 500,
  }
  return (
    <Layout style={layoutStyle}>
      <Header style={headerStyle}>header</Header>
      <Layout>
        <Content style={contentStyle}>content</Content>
        <Aside style={asideStyle}>aside</Aside>
      </Layout>
      <Footer style={footerStyle}>footer</Footer>
    </Layout>
  )
}}
</Playground>

## Layout 4

<Playground>
{() => {
  const headerStyle={
    backgroundColor:'#f4f3f3',
    padding:'1em',
  }
  const contentStyle={
    backgroundColor:'#dfdfdf',
    padding:'1em',
  }
  const asideStyle={
    backgroundColor:'#bfd8d5',
    padding:'1em',
  }
  const footerStyle = headerStyle;
  const layoutStyle = {
    height: 300,
    maxWidth: 500,
  }
  return (
    <Layout style={layoutStyle}>
      <Aside style={asideStyle}>aside</Aside>
      <Layout>
        <Header style={headerStyle}>header</Header>
        <Content style={contentStyle}>content</Content>
        <Footer style={footerStyle}>footer</Footer>
      </Layout>
    </Layout>
  )
}}
</Playground>

## Props

### Layout

<Props of={Layout} />

### Header

<Props of={Header} />

### Content

<Props of={Content} />

### Aside

<Props of={Aside} />

### Footer

<Props of={Footer} />

