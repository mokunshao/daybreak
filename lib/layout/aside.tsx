import React from 'react';
import joinedClass from '../utils/joinedClass';
import classes from '../utils/classes';

const layout = joinedClass('layout');

interface Props extends React.HtmlHTMLAttributes<HTMLElement> { }

const Aside: React.FunctionComponent<Props> = (props) => {
  const { className, children, ...rest } = props;
  return (
    <div className={classes(layout('aside'), className)} {...rest}>{children}</div>
  );
};

export default Aside;
