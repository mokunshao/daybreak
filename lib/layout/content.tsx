import React from 'react';
import joinedClass from '../utils/joinedClass';
import classes from '../utils/classes';

const layout = joinedClass('layout');

interface Props extends React.HtmlHTMLAttributes<HTMLElement> { }

const Content: React.FunctionComponent<Props> = (props) => {
  const { className, children, ...rest } = props;
  return (
    <div className={classes(layout('content'), className)} {...rest}>{children}</div>
  );
};

export default Content;
