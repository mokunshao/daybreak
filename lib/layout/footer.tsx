import React from 'react';
import { jc } from '../utils/joinedClasses';

const layout = jc('layout');

const Footer: React.FunctionComponent = () => (
  <div className={layout('footer')}>footer</div>
);

export default Footer;
