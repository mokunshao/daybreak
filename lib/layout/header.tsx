import React from 'react';
import { jc } from '../utils/joinedClasses';
import classes from '../utils/classes';

const layout = jc('layout');

interface Props extends React.HtmlHTMLAttributes<HTMLElement> { }

const Header: React.FunctionComponent<Props> = (props) => {
  const { className, children, ...rest } = props;
  return (
    <div className={classes(layout('header'), className)} {...rest}>{children}</div>
  );
};

export default Header;
