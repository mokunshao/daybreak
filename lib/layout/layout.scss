@import '../common.scss';

.daybreak-layout {
  display: flex;
  flex-direction: column;

  &-content {
    flex-grow: 1;
  }

  & & {
    flex-grow: 1;
    flex-direction: row;
  }

  &-has-aside {
    flex-direction: row;
  }

  &-has-aside & {
    flex-direction: column;
  }
}
