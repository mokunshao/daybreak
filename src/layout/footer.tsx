import React from 'react';
import joinedClass from '../utils/joinedClass';
import classes from '../utils/classes';

const layout = joinedClass('layout');

type Props = React.HtmlHTMLAttributes<HTMLElement>;

const Footer: React.FunctionComponent<Props> = (props) => {
  const { className, children, ...rest } = props;
  return (
    <div className={classes(layout('footer'), className)} {...rest}>{children}</div>
  );
};

export default Footer;
