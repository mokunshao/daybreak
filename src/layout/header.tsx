import React from 'react';
import joinedClass from '../utils/joinedClass';
import classes from '../utils/classes';

const layout = joinedClass('layout');

type Props = React.HtmlHTMLAttributes<HTMLElement>;

export const Header: React.FunctionComponent<Props> = (props) => {
  const { className, children, ...rest } = props;
  return (
    <div className={classes(layout('header'), className)} {...rest}>{children}</div>
  );
};

export default Header;
