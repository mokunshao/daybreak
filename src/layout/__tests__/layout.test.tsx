import React from 'react';
import { render } from '@testing-library/react';
import { Aside } from '../aside';
import { Content } from '../content';
import { Footer } from '../footer';
import { Layout } from '../layout';
import { Header } from '../header';

function Example() {
  return (
    <Layout>
      <Header>header</Header>
      <Layout>
        <Aside>aside</Aside>
        <Content>content</Content>
      </Layout>
      <Footer>footer</Footer>
    </Layout>
  );
}

describe('CodePreview', () => {
  it('exits', () => {
    expect(Aside).toBeTruthy();
    expect(Content).toBeTruthy();
    expect(Footer).toBeTruthy();
    expect(Layout).toBeTruthy();
    expect(Header).toBeTruthy();
  });

  it('match snapshot', () => {
    const { asFragment } = render(<Example />);
    expect(asFragment()).toMatchSnapshot();
  });
});
