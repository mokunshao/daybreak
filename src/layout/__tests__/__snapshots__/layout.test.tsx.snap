// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodePreview match snapshot 1`] = `
<DocumentFragment>
  <div
    class="daybreak-layout"
  >
    <div
      class="daybreak-layout-header"
    >
      header
    </div>
    <div
      class="daybreak-layout daybreak-layout-has-aside"
    >
      <div
        class="daybreak-layout-aside"
      >
        aside
      </div>
      <div
        class="daybreak-layout-content"
      >
        content
      </div>
    </div>
    <div
      class="daybreak-layout-footer"
    >
      footer
    </div>
  </div>
</DocumentFragment>
`;
